test/GroupByTest.cpp

```diff
@@ -421,7 +445,100 @@ TEST_F(GroupBySpecialCount, getPermutationForThreeVariableTriple) {
 }
 
 // _____________________________________________________________________________
-TEST_F(GroupBySpecialCount, checkIfJoinWithFullScan) {
+TEST_F(GroupByOptimizations, checkIfHashMapOptimizationPossible) {
+  auto testFailure = [this](const auto& groupByVariables, const auto& aliases,
+                            const auto& join, const auto& aggregates) {
+    auto groupBy = GroupBy{qec, groupByVariables, aliases, join};
+    ASSERT_FALSE(groupBy.checkIfHashMapOptimizationPossible(aggregates));
+  };
+
+  std::vector<Variable> variablesXAndY{varX, varY};
+
+  std::vector<ColumnIndex> sortedColumnsVector = {0};
+  Tree validJoinWhenGroupingByXPlusSort = makeExecutionTree<Sort>(
+      qec, validJoinWhenGroupingByX, sortedColumnsVector);
+
+  SparqlExpressionPimpl avgXPimpl = makeAvgPimpl(varX);
+  SparqlExpressionPimpl avgCountXPimpl = makeAvgCountPimpl(varX);
+
+  std::vector<Alias> aliasesAvgX{Alias{avgXPimpl, Variable{"?avg"}}};
+  std::vector<Alias> aliasesAvgCountX{
+      Alias{avgCountXPimpl, Variable("?avgcount")}};
+
+  std::vector<GroupBy::Aggregate> countAggregate = {{countXPimpl, 1}};
+  std::vector<GroupBy::Aggregate> avgAggregate = {{avgXPimpl, 1}};
+  std::vector<GroupBy::Aggregate> avgCountAggregate = {{avgCountXPimpl, 1}};
+
+  // Must have exactly one variable to group by.
+  testFailure(emptyVariables, aliasesAvgX, validJoinWhenGroupingByXPlusSort,
+              avgAggregate);
+  testFailure(variablesXAndY, aliasesAvgX, validJoinWhenGroupingByXPlusSort,
+              avgAggregate);
+  // Must be AVG aggregate (for now)
+  testFailure(variablesOnlyX, aliasesCountX, validJoinWhenGroupingByXPlusSort,
+              countAggregate);
+  // Top operation must be SORT
+  testFailure(variablesOnlyX, aliasesAvgX, validJoinWhenGroupingByX,
+              avgAggregate);
+  // Can not be a nested aggregate
+  testFailure(variablesOnlyX, aliasesAvgCountX,
+              validJoinWhenGroupingByXPlusSort, avgCountAggregate);
+  // Optimization has to be enabled
+  RuntimeParameters().set<"use-group-by-hash-map-optimization">(false);
```

Comment:
Maybe explicitly set this to `true` before your first test, then these tests do not rely on the default value here.